include/deal.II/lac/trilinos_tpetra_vector.templates.h

```diff
@@ -63,9 +63,17 @@ namespace LinearAlgebra
       : Subscriptor()
       , compressed(V.compressed)
       , has_ghost(V.has_ghost)
-      , vector(V.vector)
-      , nonlocal_vector(V.nonlocal_vector)
-    {}
+      , vector(Utilities::Trilinos::internal::make_rcp<VectorType>(
+          V.vector->getMap()))
+    {
+      Tpetra::deep_copy(*vector, *V.vector);
```

Comment:
Why do we wrap the `VectorType`? AFAIK `Tpetra::Vector` has `View`-semantics so it's already reference-counted and does shallow copies by default.

Reply:
> Why not just `vector(V.vector, Techos::DataAccess::Copy)`?

I took a look into Trilinos itself and found this solution.
I don't have a preference.

Reply:
I guess the solution indicated by @masterleinad would avoid first setting the elements to zero and then copying them, which is preferrable.

Reply:
> I guess the solution indicated by @masterleinad would avoid first setting the elements to zero and then copying them, which is preferrable.

I have not thought about this. So let's use `vector(V.vector, Techos::DataAccess::Copy)` here.

Reply:
As discussed in #16549, I do not think it is worth dropping the `Teuchos::RCP` to simplify this function. In the long run, dropping the `Teuchos::RCP` would cause more problems than it solves.